src/admin.tsx

```diff
@@ -1,30 +1,101 @@
-import React, { useContext } from 'react';
+import React, { useContext, useEffect, useState } from 'react';
 import './index.css';
 import { AuthContext } from "./AuthProvider";
-import { Button } from '@mui/material';
+import { Button, ButtonGroup } from '@mui/material';
+import { collection, getFirestore, onSnapshot, DocumentData } from "firebase/firestore";
+import AppBar from '@mui/material/AppBar';
+import Box from '@mui/material/Box';
+import Toolbar from '@mui/material/Toolbar';
+import Typography from '@mui/material/Typography';
+import ExpandMoreIcon from '@mui/icons-material/ExpandMore';
+import ChevronRightIcon from '@mui/icons-material/ChevronRight';
+import { TreeView } from '@mui/x-tree-view/TreeView';
+import { TreeItem } from '@mui/x-tree-view/TreeItem';
+
 
 export default function AdminPage() {
+  const db = getFirestore();
   const { currentUser, signOut } = useContext(AuthContext);
+  const [data, setData] = useState<DocumentData[]>();
```

Comment:
Instead of using DocumentData here lets create a custom TS Type which properly matches our Data structure within out DB. As of now this is just typed as:
```
export declare interface DocumentData {
    /** A mapping between a field and its value. */
    [field: string]: any;
}
```

On line 23 we will need to cast it then:

```
snapshot.docs.map(doc => doc.data() as NewType)
```

Reply:
This will help prevent bugs as our TS will yell at us if we are using it improperly